data_override/data_override.F90

```diff
@@ -605,6 +633,84 @@ subroutine get_domainUG(gridname, UGdomain, comp_domain)
      call mpp_get_compute_domain(SGdomain,comp_domain(1),comp_domain(2),comp_domain(3),comp_domain(4))
 end subroutine get_domainUG
 !===============================================================================================
+! <FUNCTION NAME="new_nudge3D">
+!   <DESCRIPTION>
+! This function constructs a 3D data_nudge object.
+!   </DESCRIPTION>
+function new_nudge3D(newgridname, newvarname, xbounds, ybounds, zbounds, newobscoeffname, newmodcoeffname)
+
+   character(len=3),           intent(in) :: newgridname
+   character(len=*),           intent(in) :: newvarname
+   integer, dimension(2),      intent(in) :: xbounds, ybounds, zbounds
+   character(len=*), optional, intent(in) :: newobscoeffname
+   character(len=*), optional, intent(in) :: newmodcoeffname
+
+   type(nudge3D)                          :: new_nudge3D
+ 
+   new_nudge3D%gridname = newgridname
+   new_nudge3D%varname = newvarname
+   new_nudge3D%obs_coefficient_name = trim(new_nudge3D%varname) // '_coeff'
+   if(present(newobscoeffname)) then
+      new_nudge3D%obs_coefficient_name = newobscoeffname
+   endif
+   new_nudge3D%mod_coefficient_name = trim(new_nudge3D%varname) // '_mod_coeff'
+   if(present(newmodcoeffname)) then
+      new_nudge3D%mod_coefficient_name = newmodcoeffname
+   endif
+   allocate(new_nudge3D%model_input(xbounds(1):xbounds(2),ybounds(1):ybounds(2),zbounds(1):zbounds(2)))
+   allocate(new_nudge3D%nudge_output(xbounds(1):xbounds(2),ybounds(1):ybounds(2),zbounds(1):zbounds(2)))
+   allocate(new_nudge3D%observed_data(xbounds(1):xbounds(2),ybounds(1):ybounds(2),zbounds(1):zbounds(2)))
+   allocate(new_nudge3D%observation_coefficient(xbounds(1):xbounds(2),ybounds(1):ybounds(2),zbounds(1):zbounds(2)))
+   allocate(new_nudge3D%model_coefficient(xbounds(1):xbounds(2),ybounds(1):ybounds(2),zbounds(1):zbounds(2)))
+
+end function new_nudge3D
+!===============================================================================================
+! <SUBROUTINE NAME="data_nudge_3d">
+!   <DESCRIPTION>
+! This routine performs data nudge for a variable, its observations, and the associated coefficients.
+!   </DESCRIPTION>
```

Comment:
These comments should be in doxygen format

Reply:
Changed to Doxygen (comments welcome)